### PR TITLE
Add logits_processor parameter, used by `generate`, to `Seq2SeqTrainer` methods `evaluate` and `predict`

### PR DESCRIPTION
**Pull request by eranhirs**
# What does this PR do?

Following the discussion in #17748, this PR adds the `logits_processor` param to Seq2SeqTrainer `predict` and `evaluate` methods, for easy extensibility.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@gante @sgugger

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! I think it would be better to just be done with it and store all kwargs for `generate`.

File: src/transformers/trainer_seq2seq.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def evaluate(
         metric_key_prefix: str = "eval",
         max_length: Optional[int] = None,
         num_beams: Optional[int] = None,
+        logits_processor: Optional[LogitsProcessorList] = None,
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/transformers/trainer_seq2seq.py]: I meant as **kwargs, not as the whole list :-)

[the remark this answers is not in the document]
Comment (eranhirs) [reply src/transformers/trainer_seq2seq.py]: Thanks for the feedback! Done, please review again :-)

**Comment by eranhirs** [comment]
@sgugger What are your thoughts about adding to the `test_finetune_bert2bert` a call to `trainer.evaluate()` and `trainer.predict(eval_dataset)`?

**Comment by sgugger** [comment]
Sure, we can add this. But is it to test any of this functionality? If not, it should go in its own PR.

**Review by sgugger — APPROVED** [review]

**Review by gante — APPROVED** [review]
LGTM 👍

**Comment by sgugger** [comment]
Merging this one then, thanks again for your contribution!

**Pull request merged**